app/repository/helpers.py

```diff
@@ -10,40 +10,62 @@
 from sqlalchemy.orm import Session
 
 
-def generate_slug(
-    db: Session,
-    title: str,
-    attempts: int = 100,
-    suffix_length: int = 4,
+def generate_unique_slug(
+    existing_slugs: set[str], title: str, attempts: int = 100, suffix_length: int = 6
 ) -> str:
     """
-    Generates a slug that doesn't already exists.
+    Generates a slug that doesn't already exist.
 
-    :param Session db: The connection to the db
-    :param str title: The title or name of the object you wish to slugify
+    :param set[str] existing_slugs: A set of already existing slugs to validate against.
+    :param str title: The title or name of the object you wish to slugify.
     :param int attempts: The number of attempt to generate a unique slug before
-    failing, defaults to 100
-    :param int suffix_length: The suffix to produce uniqueness, defaults to 4
-    :raises RuntimeError: If we cannot produce a unique slug
-    :return str: the slug generated
+    failing, defaults to 100.
+    :param int suffix_length: The suffix to produce uniqueness, defaults to 6.
+    :raises RuntimeError: If we cannot produce a unique slug.
+    :return str: The slug generated.
     """
-    lookup = set([cast(str, n) for n in db.query(Slug.name).all()])
     base = slugify(str(title))
 
     # TODO: try to extend suffix length if attempts are exhausted
     suffix = str(uuid4())[:suffix_length]
     count = 0
-    while (slug := f"{base}_{suffix}") in lookup:
+    while (slug := f"{base}_{suffix}") in existing_slugs:
         count += 1
         suffix = str(uuid4())[:suffix_length]
         if count > attempts:
             raise RuntimeError(
                 f"Failed to generate a slug for {base} after {attempts} attempts."
             )
-    lookup.add(slug)
+    existing_slugs.add(slug)
     return slug
 
 
+def generate_slug(
+    db: Session,
+    title: str,
+    attempts: int = 100,
+    suffix_length: int = 4,
+    created_slugs: Optional[set[str]] = set(),
+) -> str:
+    """
+    Generates a slug for a given title.
+
+    :param Session db: The connection to the db
+    :param str title: The title or name of the object you wish to slugify
+    :param int attempts: The number of attempt to generate a unique slug before
+    failing, defaults to 100
+    :param int suffix_length: The suffix to produce uniqueness, defaults to 4
+    :param Optional[set[str]] created_slugs: A set of slugs created in the context within which
+    this function runs that have not yet been committed to the DB
+    :return str: the slug generated
+    """
+    saved_slugs = set([cast(str, n) for n in db.query(Slug.name).all()])
```

Comment:
Probably worth a note here - to improve this code / reference to a ticket

Reply:
Can you clarify what you mean here please :) ?